src/nat/tool/datetime_tools.py

```diff
@@ -21,22 +21,57 @@
 
 class CurrentTimeToolConfig(FunctionBaseConfig, name="current_datetime"):
     """
-    Simple tool which returns the current date and time in human readable format.
+    Simple tool which returns the current date and time in human readable format with timezone information. By default,
+    the timezone is in Etc/UTC. If the user provides a timezone in the header, we will use it. Timezone will be
+    provided in IANA zone name format. For example, "America/New_York" or "Etc/UTC".
     """
     pass
 
 
 @register_function(config_type=CurrentTimeToolConfig)
-async def current_datetime(config: CurrentTimeToolConfig, builder: Builder):
+async def current_datetime(_config: CurrentTimeToolConfig, _builder: Builder):
 
     import datetime
+    import zoneinfo
+
+    from starlette.datastructures import Headers
+
+    from nat.settings.global_settings import GlobalSettings
 
     async def _get_current_time(unused: str) -> str:
 
-        now = datetime.datetime.now()  # Get current time
-        now_human_readable = now.strftime(("%Y-%m-%d %H:%M:%S"))
+        del unused  # Unused parameter to avoid linting error
+
+        from nat.builder.context import Context
+        nat_context = Context.get()
+
+        # Default to UTC
+        timezone_obj = zoneinfo.ZoneInfo("Etc/UTC")
```

Comment:
This seems a bit like a misnomer to have the default be UTC when the fallback is definable.